src/sinks/aws_sqs.rs

```diff
@@ -210,12 +210,7 @@ impl Service<Vec<SendMessageEntry>> for SqsSink {
         Box::pin(async move {
             client
                 .send_message(request)
-                .inspect_ok(|result| {
-                    emit!(AwsSqsEventSent {
-                        byte_size,
-                        message_id: result.message_id.as_ref()
-                    })
-                })
+                .map_ok(|result| (result, byte_size))
```

Comment:
So this pattern actually seems fine as it was, right? We are sending the request and then record the metric once that has succeeded.

Does that mean this is only a problem for `HttpSink`? It'd be nice if we could limit the change to that trait.

Reply:
It is not. Partially this pattern was correct, but because this service wrapped by retry layer (we have retry logic for any batch sink) we will emit metric even if we will retry the request what means that events/bytes will be counted few times.